hadoop/src/main/java/org/apache/carbondata/hadoop/api/CarbonInputFormat.java

```diff
@@ -573,19 +573,27 @@ private int getBlockCount(List<ExtendedBlocklet> blocklets) {
       if (cgDataMapExprWrapper != null) {
         // Prune segments from already pruned blocklets
         DataMapUtil.pruneSegments(segmentIds, prunedBlocklets);
-        List<ExtendedBlocklet> cgPrunedBlocklets;
+        List<ExtendedBlocklet> cgPrunedBlocklets = new ArrayList<>();
         // Again prune with CG datamap.
         if (distributedCG && dataMapJob != null) {
-          cgPrunedBlocklets = DataMapUtil
-              .executeDataMapJob(carbonTable, filter.getResolver(), dataMapJob, partitionsToPrune,
-                  segmentIds, invalidSegments, DataMapLevel.CG, true, new ArrayList<String>());
+          try {
+            cgPrunedBlocklets = DataMapUtil
+                .executeDataMapJob(carbonTable, filter.getResolver(), dataMapJob, partitionsToPrune,
+                    segmentIds, invalidSegments, DataMapLevel.CG, true, new ArrayList<String>());
+          } catch (Exception e) {
+            LOG.error("CG datamap pruning failed.", e);
+          }
         } else {
           cgPrunedBlocklets = cgDataMapExprWrapper.prune(segmentIds, partitionsToPrune);
         }
-        // since index datamap prune in segment scope,
-        // the result need to intersect with previous pruned result
-        prunedBlocklets =
-            intersectFilteredBlocklets(carbonTable, prunedBlocklets, cgPrunedBlocklets);
+        // If cgPrunedBlocklets == 0, it means that CG datamap pruning failed,
```

Comment:
an empty collection could also mean all blocklet are filtered and no data hit

Reply:
@kevinjmh Case 1: If there is no data corresponding to the filter column, then simply it will pass the pruned blocklets from default datamap and will not go for cg pruning.
Case 2: If there is data corresponding to filter column, then cg pruned blocklets should not be 0 and if it is 0, it means that there is some problem in cg pruning, So we do not fail the select query and fallback to default pruning.

Reply:
For case 2, let's say some blocklet has a column with data lying as `1,3,5,7,9` and its min max value are 1 and 9.  If here comes a filter `col=2`, the default datamap found 2 is between 1 and 9, and take this blocklet to scan. Then it goes to cg pruning, it cound be found that value 2 does not exist in blocklet using hash by bloom datamap.

Reply:
@kevinjmh Handled this scenario...please review.